docs/features/traffic-management/istio.md

```diff
@@ -1,3 +1,88 @@
 # Istio
 
-TBD
\ No newline at end of file
+NOTE: This functionality will be available in the v0.7 version, and everything described below is subject to change.
+
+[Istio](https://istio.io/) is one of the most popular service mesh in the community and offers a rich feature-set to control the flow of traffic. Istio offers this functionality through a set of CRDs, and the Argo Rollouts controller modifies these resources to manipulate the traffic routing into the desired state. However, the Argo Rollouts controller modifies the Istio resources minimally to gives the developer flexibility while configuring their resources.
+
+## Istio and Rollouts
+
+The Argo Rollout controller achieves traffic shaping by manipulating the Virtual Service. A Virtual Service provides the flexibility to define how to route traffic when a host address is hit.  The Argo Rollouts controller manipulates the Virtual Service by using the following Rollout configuration:
+
+- Canary Service name
+- Stable Service name
+- Virtual Service Name
+- Which HTTP Routes in the Virtual Service
+
+Below is an example of a Rollout with all the required fields configured:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Rollout
+metadata:
+  name: rollout-example
+spec:
+  ...
+  strategy:
+    canary:
+      steps:
+      - setWeight: 5
+      - pause:
+          duration: 600
+      canaryService: canary-svc # required
+      stableService: stable-svc  # required
+      networking:
```

Comment:
Not totally sold on the `networking` field name. Probably needs the word traffic in it. But we can discuss later.

Reply:
Maybe `trafficSplit`:

```yaml
spec:
  strategy:
    canary:
      trafficSplit:
        smi: {}
```


```yaml
spec:
  strategy:
    canary:
      trafficSplit:
        istio:
...
```

Reply:
Yeah, I agree that we can find something better than `networking`. I'm wondering if `trafficSplit` is too specific since this configuration will be also be used for traffic mirroring and header-based routing

Reply:
What about `trafficRouting`?